### PR TITLE
TOPIC

### PR DESCRIPTION
**Pull request by djames9**
the nick now does not accept the , plus fix the syntax what irssi expect in topic and done actionTopic. there is one question for one case. we need to decide together. according to that decision we delete or uncomment the last line of the function. llets go

### CODE REVIEW

File: src/common/reply.h [hunk 1]
```diff
@@ -18,7 +18,8 @@
     (RPL_META_MESSAGE(servername, "004", nick + " " + servername + " " + version + " " + user_modes + " " + channel_modes))
 
 #define RPL_NOTOPIC_331(servername, channel) (RPL_META_MESSAGE(servername, "331", channel + " :No topic is set"))
-#define RPL_TOPIC_332(servername, channel, topic) (RPL_META_MESSAGE(servername, "332", channel + " :" + topic))
+#define RPL_TOPIC_332(nick, user, servername, channel, topic) \
+    (RPL_META_MESSAGE(nick + "!" + user + "@" + servername, "332 TOPIC", channel + " " + topic))
```
Comment (mxafi) [inline src/common/reply.h:22]: Reference | Content after `:nick!user@serverHost 332 ` in `RPL_TOPIC`
--- | ---
Per IRCv3 | `<client> <channel> :<topic>`
Per RFC2812 | `<channel> :<topic>`

Note: `RPL_NOTOPIC` is following RFC2812

So the current implementation in this PR seems incorrect.

**Review by mxafi — CHANGES_REQUESTED** [review]
I left individual comments on the issues I found, please fix 👀

File: src/common/reply.h (hunk 1 above)
Comment (djames9) [reply src/common/reply.h]: change

File: src/command/Command.cpp [hunk 2]
```diff
@@ -219,6 +223,47 @@ void Command::actionPart(Client& client) {
...
+            it->second.setTopic(param_.back());
+            LOG_DEBUG(topic);
+            it->second.sendMessageToMembers(
+                RPL_TOPIC_332(client.getNickname(), client.getUserName(), serverHostname_g, param_.at(0), it->second.getTopic()));
+            return;
+        }
+        //client.appendToSendBuffer(RPL_ERR_NEEDMOREPARAMS_461(serverHostname_g, "TOPIC"));
+        return;
+    }
+}
```
[the remark this answers is not in the document]
Comment (djames9) [reply src/command/Command.cpp:266]: change

File: src/command/Command.cpp [hunk 3]
```diff
@@ -255,13 +255,14 @@ void Command::actionTopic(Client& client) {
         if (topic[0] == ':') {
             it->second.setTopic(param_.back());
```
Comment (mxafi) [inline src/command/Command.cpp:256]: Topic is being set straight to the parameter.. it should strip the colon beforehand.
Also if the parameter is only a colon, it should set the topic to an empty string.

File: src/command/Command.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (mxafi) [reply src/command/Command.cpp:266]: Also the logical flow still has some polishing to do:

- it is missing the case for more than 2 parameters, it is currently replying needmoreparams if we get more than 2... a desirable implementation would be that we handle any amount of parameters without errors after 2, we just dont use them
- The topic should be viewable only by channel members (not a hard requirement by rfc2812, but a recommendation by ircv3): _This could also be used to optimise the allChannels.find, as seen in the example flow in my first comment in this thread_

**Review by mxafi — CHANGES_REQUESTED** [review]
There's still some polishing to meet RFC2812, IRCv3, subject, and my requirements!

Please re-read TOPIC command and MODE command requirements from all of the above, as well as my comment about the TOPIC function's flow

**Pull request merged**